### PR TITLE
Delete user from token

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR adds a DELETE method to the `fromtoken` endpoint, so that UI callers can delete Auth0 accounts during the setup process (i.e. before the record in Mongo is created).

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/auth/Auth0Connection.java [hunk 1]
```diff
@@ -100,7 +101,7 @@ public static void checkUser(Request req) {
      * @return true if the request can operate without a user profile, false otherwise.
      */
     private static boolean expectsMissingProfile(Request req, RequestingUser profile) {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/auth/Auth0Connection.java:104]: The comment on this method should probably be updated and perhaps also the log message on line 78.

File: src/main/java/org/opentripplanner/middleware/controllers/api/AbstractUserController.java [hunk 2]
```diff
@@ -98,6 +106,45 @@ private U getUserFromRequest(Request req, Response res) {
...
+     * HTTP endpoint to delete the {@link U} entity, if it exists, from an {@link RequestingUser} token
+     * available from a {@link Request} (this is the case for '/api/secure/' endpoints).
+     */
+    private boolean deleteUserFromRequest(Request req, Response res) {
+        RequestingUser profile = Auth0Connection.getUserFromRequest(req);
+        if (profile == null) {
+            logMessageAndHalt(req,
+                HttpStatus.NOT_FOUND_404,
+                String.format(NO_USER_WITH_AUTH0_ID_MESSAGE, null),
+                null);
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/controllers/api/AbstractUserController.java]: What formatting are you using? This and a few other lines wrap were I wouldn't expect.

**Review by br648 — APPROVED** [review]
Just a couple of minors, but all good.

File: src/main/java/org/opentripplanner/middleware/controllers/api/AbstractUserController.java (hunk 2 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/controllers/api/AbstractUserController.java]: I copied stuff over from `getUserFromRequest`, so I guess the formatting comes from there... I agree that `req` should be on its own line.

File: src/main/java/org/opentripplanner/middleware/auth/Auth0Connection.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/auth/Auth0Connection.java:104]: Thanks, updated in https://github.com/ibi-group/otp-middleware/pull/213/commits/d73eb97328616713dbe7d034f4f3e926434ab291.

File: src/main/java/org/opentripplanner/middleware/controllers/api/AbstractUserController.java (hunk 2 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/controllers/api/AbstractUserController.java]: Refactored in https://github.com/ibi-group/otp-middleware/pull/213/commits/af8b9324ec3b3da4b4fab50380bb353795b734cc.

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Code looks good to me! Not experienced with the middleware though.

**Pull request merged**